filemanager/digitaloceanmanager_v2.go

```diff
@@ -0,0 +1,328 @@
+package filemanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/smithy-go"
+	"github.com/mitchellh/mapstructure"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+type DigitalOceanConfigV2 struct {
+	Bucket         string
+	Prefix         string
+	EndPoint       string
+	AccessKeyID    string
+	AccessKey      string
+	Region         *string
+	ForcePathStyle *bool
+	DisableSSL     *bool
+}
+
+type DigitalOceanManagerV2 struct {
+	*baseManager
+	config *DigitalOceanConfigV2
+
+	sessionConfig *awsutil.SessionConfig
+	client        *s3.Client
+	clientMu      sync.Mutex
+}
+
+func NewDigitalOceanManagerV2(
+	config map[string]interface{}, log logger.Logger, defaultTimeout func() time.Duration,
+) (*DigitalOceanManagerV2, error) {
+	var doConfig DigitalOceanConfigV2
+	if err := mapstructure.Decode(config, &doConfig); err != nil {
+		return nil, fmt.Errorf("failed to decode DigitalOcean config: %w", err)
+	}
+
+	sessionConfig, err := awsutil.NewSimpleSessionConfig(config, "digitalocean")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create DigitalOcean session config: %w", err)
+	}
+
+	// DigitalOcean Spaces requires a region, but it's often embedded in the endpoint
+	if doConfig.Region == nil || *doConfig.Region == "" {
+		region := getSpacesLocationV2(doConfig.EndPoint)
+		doConfig.Region = &region
+		sessionConfig.Region = region
+	}
+
+	return &DigitalOceanManagerV2{
+		baseManager: &baseManager{
+			logger:         log,
+			defaultTimeout: defaultTimeout,
+		},
+		config:        &doConfig,
+		sessionConfig: sessionConfig,
+	}, nil
+}
+
+func (m *DigitalOceanManagerV2) getClient(ctx context.Context) (*s3.Client, error) {
+	m.clientMu.Lock()
+	defer m.clientMu.Unlock()
+
+	if m.client != nil {
+		return m.client, nil
+	}
+
+	if m.config.Bucket == "" {
+		return nil, errors.New("no storage bucket configured to downloader")
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, m.getTimeout())
+	defer cancel()
+
+	cnf, err := awsutil.CreateAWSConfig(ctx, m.sessionConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create DigitalOcean AWS config: %w", err)
+	}
+
+	client := s3.NewFromConfig(cnf, func(o *s3.Options) {
+		if m.config.EndPoint != "" {
+			o.BaseEndpoint = aws.String(m.config.EndPoint)
```

Comment:
This can be both `http` and `https`  and it should be defined by the consumer of this as to whether it is http or https.